Modules/xSharePoint/DSCResources/MSFT_xSPShellAdmins/MSFT_xSPShellAdmins.psm1

```diff
@@ -0,0 +1,584 @@
+                                } catch {
+                                    throw "Error while setting the Shell Admin. The Shell Admin permissions will not be applied. Error details: $($_.Exception.Message)"
+                                    return
+                                }
+                            }
+                        }
+                    }
+
+                    if ($contentDatabase.MembersToInclude) {
+                        Write-Verbose -Verbose "Processing MembersToInclude"
+                        if ($dbShellAdmins) {
+                            ForEach ($member in $contentDatabase.MembersToInclude) {
+                                if (-not $dbShellAdmins.UserName.Contains($member)) {
+                                    try {
+                                        Add-SPShellAdmin -database $currentCDB.Id -UserName $member
+                                    } catch {
+                                        throw "Error while setting the Shell Admin. The Shell Admin permissions will not be applied. Error details: $($_.Exception.Message)"
+                                        return
+                                    }
+                                }
+                            }
+                        } else {
+                            ForEach ($member in $contentDatabase.MembersToInclude) {
+                                try {
+                                    Add-SPShellAdmin -database $currentCDB.Id -UserName $member
+                                } catch {
+                                    throw "Error while setting the Shell Admin. The Shell Admin permissions will not be applied. Error details: $($_.Exception.Message)"
+                                    return
+                                }
+                            }
+                        }
+                    }
+
+                    if ($contentDatabase.MembersToExclude) {
+                        Write-Verbose -Verbose "Processing MembersToExclude"
+                        if ($dbShellAdmins) {
+                            ForEach ($member in $contentDatabase.MembersToExclude) {
+                                if ($dbShellAdmins.UserName.Contains($member)) {
+                                    try {
+                                        Remove-SPShellAdmin -database $currentCDB.Id -UserName $member -Confirm:$false
+                                    } catch {
+                                        throw "Error while removing the Shell Admin. The Shell Admin permissions will not be revoked. Error details: $($_.Exception.Message)"
+                                        return
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    throw "Specified database does not exist"
```

Comment:
Can we include the name of the database it was looking for in this error? That way if we specify multiple databases we know which one is at fault